addons/purchase_stock/models/account_invoice.py

```diff
@@ -133,9 +133,12 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
                                 'tax_ids': tax_ids,
                             }
                             # We update the original line accordingly
-                            line['price_unit'] = line['price_unit'] - diff_line['price_unit']
-                            line['price'] = inv.currency_id.round(line['quantity'] * line['price_unit'])
-                            line['price_unit'] = inv.currency_id.round(line['price_unit'])
+                            # line['price_unit'] doesn't contain the discount, so use price_unit
+                            # instead. It could make sense to include the discount in line['price_unit'],
+                            # but that doesn't seem a good idea in stable since it is done in
+                            # "invoice_line_move_line_get" of "account.invoice".
+                            line['price_unit'] = inv.currency_id.round(price_unit - diff_line['price_unit'])
```

Comment:
@nim-odoo I know this isn't in the original report, but look at line 118 and also 99. It's got the same price_unit problem, not taking into account discount.  I guess the test for that is a product valued at 90, and comparing the resulting journal entries for an invoice with 90 price unit, with an invoice with 100 price unit and a 10% discount. They should be identical.

Reply:
Yeah I know, I hesitated to change it as well but that would make sense.

Reply:
It should be good now.